test/unit/io/alignment_file/alignment_file_format_test_template.hpp

```diff
@@ -545,7 +547,7 @@ TYPED_TEST_P(alignment_file_write, cigar_vector)
     }
 
     this->ostream.flush();
-    EXPECT_EQ(this->ostream.str(), this->simple_three_reads_output);
+    EXPECT_EQ(this->ostream.str(), this->simple_three_reads_input);
```

Comment:
Why this change to input? You want to write the alignment file, so it should be output...

Reply:
Because in the case that I write an alignment directly, or I read directly write, hard clipping is not preserved in the cigar string (because it cannot be represented in a pair of aligned sequences). In the case of this test, hard clipping is preserved when writing because I write a cigar vector (not an alignment), so I need to compare to the original input instead of the output string.

I will drop a note in the test :)

Reply:
Thanks 👍